oneflow/core/common/steady_vector.h

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 The OneFlow Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#ifndef ONEFLOW_CORE_COMMON_STEADY_VECTOR_H_
+#define ONEFLOW_CORE_COMMON_STEADY_VECTOR_H_
+
+#include <vector>
+#include <array>
+#include <mutex>
+#include <cmath>
+#include <glog/logging.h>
+
+namespace oneflow {
+
+template<typename T, int N = 20>
+class SteadyVector {
+ public:
+  SteadyVector() : size_(0) {}
+  ~SteadyVector() = default;
+
+  using value_type = const T;
+  using size_type = size_t;
+
+  // thread safe.
+  size_t size() const { return size_; }
+
+  // thread safe.
+  const T& at(size_t index) const {
+    CHECK_GE(index, 0);
+    CHECK_LT(index, size_);
+    int gran = GetGranularity(index);
+    int start = (1 << gran) - 1;
+    return granularity2vector_[gran].data()[index - start];
```

Comment:
这里可以直接调用 operator[] 了，比如
```suggestion
    return operator[](index);
```

Reply:
已处理：https://github.com/Oneflow-Inc/oneflow/commit/e7ae284ed03e6dbbcf1bd1241863fb1bfe480be7